interfaces/builtin/serial_port.go

```diff
@@ -50,37 +59,83 @@ func (iface *SerialPortInterface) SanitizeSlot(slot *interfaces.Slot) error {
 		panic(fmt.Sprintf("slot is not of interface %q", iface))
 	}
 
+	// Allow the core snap to create implicits serial-port slots with no
+	// attributes. These will be used to grant access based on Udev rules
+	if slot.Snap.Type == "os" && len(slot.Attrs) == 0 {
+		return nil
+	}
+
+	if len(slot.Attrs) > 1 {
+		return fmt.Errorf("serial-port slot definition has unexpected number of attributes")
+	}
```

Comment:
Shouldn't this be '> 2'? AIUI, we want to be able to specify:
- path
- vendor
- product
- vendor + product